src/ios/FirebaseAnalyticsPlugin.m

```diff
@@ -101,21 +101,30 @@ - (void)requestTrackingAuthorization:(CDVInvokedUrlCommand *)command {
 - (void)showTrackingAuthorizationPopup:(CDVInvokedUrlCommand *)command {
     
     if (@available(iOS 14, *)) {
-        [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
-            BOOL result = false;
-            switch(status) {
-                case ATTrackingManagerAuthorizationStatusAuthorized: {
-                    result = true;
-                    break;
-                }
-                default: {
-                    result = false;
-                    break;
+        
+        NSDictionary *dict = NSBundle.mainBundle.infoDictionary;
+        
+        if([dict objectForKey:@"NSUserTrackingUsageDescription"]){
```

Comment:
you can join both `if` clauses with a `&&` operator as both `else`s are the same.

Reply:
not sure I should do that, xcode is giving me this warning when I join both clauses: 
![image](https://user-images.githubusercontent.com/27646996/167462828-9e71b4dd-edf9-414b-b319-dea1c696d261.png)

Reply:
Objective-C... There are other methods that do the same, but I guess it's ok to leave it as it is.

Reply:
We can still remove the duplicated code related to `CDVPluginResult` if we introduce something like the following. 
![image](https://user-images.githubusercontent.com/97543217/167576783-3b5d545e-4d54-42e4-9994-f1506f7c7ee8.png)

Reply:
Good idea! Done